File: app/src/main/java/com/erebor/tomkins/pos/repository/network/RetrofitModule.java
package com.erebor.tomkins.pos.repository.network;

import com.erebor.tomkins.pos.repository.network.TomkinsService;
import com.erebor.tomkins.pos.tools.SharedPrefs;
import com.google.gson.Gson;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

@Module
public class RetrofitModule {


    @Provides
    @Singleton
    Retrofit provideRetrofit(Gson gson, OkHttpClient okHttpClient, SharedPrefs sharedPrefs) {
        return new Retrofit.Builder()
                .baseUrl(sharedPrefs.getHostname())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .client(okHttpClient)
                .addConverterFactory(GsonConverterFactory.create(gson))
                .build();
    }

    @Provides
    @Singleton
    LoginRepository provideLoginRepository(Retrofit retrofit) {
        return new LoginRepository(retrofit.create(TomkinsService.class));
    }
}

File: app/src/main/java/com/erebor/tomkins/pos/di/RetrofitModule.kt
package com.erebor.tomkins.pos.di

import com.erebor.tomkins.pos.tools.SharedPrefs
import com.google.gson.Gson

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

@Module
class RetrofitModule {


    @Provides
    @Singleton
    internal fun provideRetrofit(gson: Gson, okHttpClient: OkHttpClient, sharedPrefs: SharedPrefs): Retrofit {
        return Retrofit.Builder()
                .baseUrl(sharedPrefs.hostname)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .client(okHttpClient)
                .addConverterFactory(GsonConverterFactory.create(gson))
                .build()
    }
    //
    //    @Provides
    //    @Singleton
    //    UserAuthService providesUserAuthService(Retrofit retrofit) {
    //        return retrofit.create(UserAuthService.class);
    //    }
}
